### PR TITLE
fix jaegerremote parser to use jsonpb unmarshaler

### PR DESCRIPTION
**Pull request by tonybase**
Failed to parse sampling strategy response: 
json: cannot unmarshal string into Go struct field SamplingStrategyResponse.strategyType of type api_v2.SamplingStrategyType:
```json
{"strategyType":"PROBABILISTIC","probabilisticSampling":{"samplingRate":1},"operationSampling":{"defaultSamplingProbability":1,"defaultLowerBoundTracesPerSecond":0,"perOperationStrategies":[{"operation":"/health","probabilisticSampling":{"samplingRate":0}},{"operation":"/metrics","probabilisticSampling":{"samplingRate":0}}}],"defaultUpperBoundTracesPerSecond":0}}

```

### CODE REVIEW

File: samplers/jaegerremote/sampler_remote.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/gogo/protobuf/jsonpb"
```
Comment (hanyuancheung) [inline samplers/jaegerremote/sampler_remote.go:29]:
```suggestion
	"github.com/gogo/protobuf/jsonpb"
	
```

**Review by hanyuancheung — APPROVED** [review]

File: samplers/jaegerremote/sampler_remote.go (hunk 1 above)
Comment (tonybase) [reply samplers/jaegerremote/sampler_remote.go:29]: Because the generate uses gogoproto, Protobuf V2 cannot be used

**Comment by hanyuancheung** [comment]
This PR should implement in your local-copy version, instead of being implemented in contrib. If you use jsonpb to Marshal, you should also copy and reimplement the jsonpb Unmarshal. But I don't think this is the standard way according to the otel specification. PTAL.

**Review by hanyuancheung — CHANGES_REQUESTED** [review]
@MrAlias how do you think about it?

**Comment by dmathieu** [comment]
This should include an automated test

**Pull request closed without merging**